File: samples/src/main/java/tk/zielony/carbonsamples/feature/ContextWrapperActivity.java
package tk.zielony.carbonsamples.feature;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;

import carbon.CarbonContextWrapper;
import tk.zielony.carbonsamples.R;

public class ContextWrapperActivity extends Activity {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_contextwrapper);
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(new CarbonContextWrapper(newBase));
    }
}

File: samples/src/main/java/tk/zielony/carbonsamples/feature/ContextWrapperActivity.kt
package tk.zielony.carbonsamples.feature

import android.content.Context
import carbon.CarbonContextWrapper
import tk.zielony.carbonsamples.ActivityAnnotation
import tk.zielony.carbonsamples.R
import tk.zielony.carbonsamples.SamplesActivity

@ActivityAnnotation(layout = R.layout.activity_contextwrapper, title = R.string.contextWrapperActivity_title)
class ContextWrapperActivity : SamplesActivity() {
    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(CarbonContextWrapper.wrap(newBase))
    }
}
